### PR TITLE
Micro-optimizations for base_wire_entity

### PR DESCRIPTION
**Pull request by Astralcircle**
Basically just saves __index calls/caches colors

### CODE REVIEW

**Review by Denneisk — APPROVED** [review]

File: lua/entities/base_wire_entity.lua [hunk 1]
```diff
@@ -40,8 +40,11 @@ if CLIENT then
 		local pos = localPly:GetEyeTrace().HitPos
 		local spos = localPly:GetShootPos()
 		if pos == spos then -- if the position is right in your face, get a better position
-			pos = spos + localPly:GetAimVector() * 5
+			pos = localPly:GetAimVector()
+			pos:Mul(5)
+			pos:Add(spos)
 		end
+
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:47]: Revert this

File: lua/entities/base_wire_entity.lua [hunk 2]
```diff
@@ -117,19 +120,19 @@ if CLIENT then
...
 			surface.DrawLine( poly[i].x, poly[i].y, poly[i+1].x, poly[i+1].y )
 		end
-		surface.DrawLine( poly[#poly].x, poly[#poly].y, poly[1].x, poly[1].y )
+		surface.DrawLine( poly[6].x, poly[6].y, poly[1].x, poly[1].y )
 	end
 
 	local function getWireName( ent )
 		local name = ent:GetNWString("WireName")
-		if not name or name == "" then return ent.PrintName else return name end
+		if not name or name == "" then return ent:GetTable().PrintName else return name end
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:135]: Revert this

File: lua/entities/base_wire_entity.lua [hunk 3]
```diff
@@ -258,11 +261,11 @@ if CLIENT then
 
 		local cur_ent = ply:GetEyeTrace().Entity
 
-		if cur_ent ~= looked_at and IsValid(looked_at) and looked_at.IsWire then
+		if cur_ent ~= looked_at and IsValid(looked_at) and looked_at:GetTable().IsWire then
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:264]: Revert this

File: lua/entities/base_wire_entity.lua [hunk 4]
```diff
@@ -258,11 +261,11 @@ if CLIENT then
...
 			looked_at:BeingLookedAtByLocalPlayer()
 		end
 
-		if IsValid(cur_ent) and cur_ent.IsWire and cur_ent:BeingLookedAtByLocalPlayer() then
+		if IsValid(cur_ent) and cur_ent:GetTable().IsWire and cur_ent:BeingLookedAtByLocalPlayer() then
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:268]: Revert this

File: lua/entities/base_wire_entity.lua [hunk 5]
```diff
@@ -282,12 +285,14 @@ if CLIENT then
...
+		if (CurTime() >= (tab.NextRBUpdate or 0)) then
 			-- We periodically update the render bounds every 10 seconds - the
 			-- reasons why are mostly anecdotal, but in some circumstances
 			-- entities might 'forget' their renderbounds. Nobody really knows
 			-- if this is still needed or not.
-			self.NextRBUpdate = CurTime() + 10
+			tab.NextRBUpdate = CurTime() + 10
 			Wire_UpdateRenderBounds(self)
 		end
 	end
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:298]: Need to investigate if this is even needed code anymore

File: lua/entities/base_wire_entity.lua [hunk 6]
```diff
@@ -301,9 +306,11 @@ if CLIENT then
 		halos_inv[self] = true
 	end
 
+	local color_halo = Color(100, 100, 255)
+
 	hook.Add("PreDrawHalos", "Wiremod_overlay_halos", function()
-		if #halos == 0 then return end
-		halo.Add(halos, Color(100,100,255), 3, 3, 1, true, true)
+		if table.IsEmpty(halos) then return end
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:312]:
```suggestion
		if halos[1]==nil then return end
```

File: lua/entities/base_wire_entity.lua [hunk 7]
```diff
@@ -315,7 +322,7 @@ if CLIENT then
 	-- Basic legacy GetOverlayText, is no longer used here but we leave it here in case other addons rely on it.
 	function ENT:GetOverlayText()
 		local name = self:GetNWString("WireName")
-		if name == "" then name = self.PrintName end
+		if name == "" then name = self:GetTable().PrintName end
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:325]: Revert this

File: lua/entities/base_wire_entity.lua [hunk 8]
```diff
@@ -373,7 +380,8 @@ function ENT:SetOverlayData( data )
 end
 
 function ENT:GetOverlayData()
-	return self.OverlayData
+	local tab = self:GetTable()
+	return tab and tab.OverlayData or nil
```
Comment (thegrb93) [inline lua/entities/base_wire_entity.lua:384]: Revert this

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
Crap code for a pinch of performance. Entity __index does GetTable itself so its not that beneficial in places where you're only avoiding one __index call

**Comment by Denneisk** [comment]
I was also questioning if `GetTable` in this usage was efficient but I assumed the best. With that said, some of these index calls are ran every tick (conditionally), and I think there's merit in having these changes there.

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**